src/index.cpp

```diff
@@ -852,9 +855,10 @@ namespace diskann {
 
       // Compute distances to unvisited nodes in the expansion
       if (_pq_dist) {
-        compute_dists(id_scratch.data(), id_scratch.size(), dist_scratch);
-
+        assert(dist_scratch.capacity() >= id_scratch.size());
+        compute_dists(id_scratch.data(), id_scratch.size(), dist_scratch.data());
```

Comment:
In this code path, `dist_scratch` is never resized, so `data()` can return a `nullptr` and result in a crash. One way to address this is to change `compute_dists` to take a reference to the `float` vector and then do `resize()` before calling `diskann::pq_dist_lookup` passing it `data()`.

Since `dist_scratch.clear()` was called in line 832, the resize would also have an effect of initializing all values in the vector to 0. The `diskann::pq_dist_lookup` method also initializes the buffer to 0, so there's double work. There are several ways to avoid that, one with minimum changes is to add a flag to `diskann::pq_dist_lookup` whether to initialize distances.